magento/magento2
# [2.2][Backport] Database Media Storage - Transactional Emails will now extract image from database in Database Media Storage mode

### Original PR (*)
1. #21674 

### Description (*)
If the email logo does not exist in pub/media and we are in database storage mode, then magento should copy the image file from database to local pub/media prior to continuing.

### Fixed Issues (if relevant)
1. magento/magento2#21671: Database Media Storage - Transaction emails logo not used when pub/media cleared

### Manual testing scenarios (*)
1. Setup Magento
2. Change media storage mode to database
3. Synchronize
4. Save Settings
5. Set store General Contact email address to valid email
6. Set email logo (Content->Configuration->Edit->Transactional Emails)
7. Clear Cache
8. Verify email logo exists in database <- This will fail on 2.2-develop due to #21672
9. Verify email logo exists in pub/media
10. On Frontend, subscribe to newsletter
11. Verify newsletter email contains logo
12. Clear pub/media, simulating a new cluster node being used
13. On Frontend, subscribe to newsletter (need to use new email address, or delete 1st address in backend first)
14. Verify newsletter email contains logo

### Contribution checklist (*)
